### PR TITLE
Nested task completion counter

### PR DESCRIPTION
**Pull request by pumbas600**
Nested items will now be created using a `TodoItemGroup` which listens to when subtasks are completed, allowing it to keep track of whether all of them have been completed, in which case, it too is considered completed. 

Screenshot showing the completion counter for nested tasks when they have all been completed:
image

Sidenotes:
* Fixed some minor bugs in `TodoItem` 
* Made `TodoList#renderItems` also create the surrounding `<ul>...</ul>`
* Consolidated all the CSS for the Todo page into one `todo.css` file (I don't think we're going to have that much CSS)
* Converted camelCase naming of classes to kebab-case

### CODE REVIEW

File: frontend/src/components/Pages/Todo/todo.css
Comment (jacobatan) [inline frontend/src/components/Pages/Todo/todo.css]: Change the hex colour code to have lowercase letters -- to keep it consistent with the rest of the CSS files.

File: frontend/src/components/Pages/Todo/TodoList.jsx
Comment (jacobatan) [inline frontend/src/components/Pages/Todo/TodoList.jsx]: The map() method takes an index as a param. I suggest we use that for the keys, instead of `index++`

**Review by jacobatan — CHANGES_REQUESTED** [review]
Other than the two very minor changes, the code looks awesome, and I'm happy to merge right after the style changes.

File: frontend/src/components/Pages/Todo/TodoList.jsx
Comment (NachoToast) [reply frontend/src/components/Pages/Todo/TodoList.jsx]: `items.length !== 0` can also be changed to `items.length` since 0 is falsy.

Comment (pumbas600) [reply frontend/src/components/Pages/Todo/TodoList.jsx]: Good points! I didn't even realise `.map((element, index) => ...)` was a thing ❤️

Comment (NachoToast) [reply frontend/src/components/Pages/Todo/TodoList.jsx]: Yeah, the `.map()` method actually has 3 parameters, `element`, `index`, and `array`, for instance a way to find out if there are duplicates in an array:
```ts
anArray = [1, 2, 3, 4, 3, 6, 7]
duplicates = anArray.map((element, index, array) => array.indexOf(element) !== index)
```

File: frontend/src/components/Pages/Todo/TodoList.jsx
Comment (NachoToast) [inline frontend/src/components/Pages/Todo/TodoList.jsx]: `todoItem.items && todoItem.items.length` could be simplified down to just `todoItem.items`

Comment (pumbas600) [reply frontend/src/components/Pages/Todo/TodoList.jsx]: What if you had an object like:
```javascript
const todoItem = {
    items: []
};
```
Wouldn't `todoItem.items` return true here, as `todoItem.items` is defined, despite it being empty?

Comment (NachoToast) [reply frontend/src/components/Pages/Todo/TodoList.jsx]: Ah mb, I meant to simplify it to `todoItem.items.length`

Comment (pumbas600) [reply frontend/src/components/Pages/Todo/TodoList.jsx]: Hmm, I'm not sure I can guarantee `items` will be defined though, but I suppose it could probably be simplified to `todoItem.items?.length`.

Comment (NachoToast) [reply frontend/src/components/Pages/Todo/TodoList.jsx]: Good idea, that will work

Comment (pumbas600) [reply frontend/src/components/Pages/Todo/TodoList.jsx]: Sweet, I'll quickly make that change :)

**Comment by pumbas600** [comment]
Before it's merged, do you think we should rename `TodoItem` and `items`, etc, to `TodoTask` and `tasks` or do you think it's fine?

**Comment by NachoToast** [comment]
> Before it's merged, do you think we should rename `TodoItem` and `items`, etc, to `TodoTask` and `tasks` or do you think it's fine?

I think its fine

**Review by NachoToast — APPROVED** [review]

**Review by jacobatan — APPROVED** [review]

**Pull request merged**
